pkg/aws/eni/node.go

```diff
@@ -452,3 +455,25 @@ func (n *Node) ResyncInterfacesAndIPs(ctx context.Context, scopedLog *logrus.Ent
 
 	return available, nil
 }
+
+// GetMaximumAllocatableIPv4 returns the maximum amount of IPv4 addresses
+// that can be allocated to the instance
+func (n *Node) GetMaximumAllocatableIPv4(scopedLog *logrus.Entry) int {
+	limits, limitsAvailable := n.getLimits()
+	if !limitsAvailable {
+		scopedLog.Warningf("Could not determined instance limits, %s", getMaximumAllocatableIPv4FailureWarningStr)
+		return 0
+	}
+
+	if limits.Adapters < *n.k8sObj.Spec.ENI.FirstInterfaceIndex {
+		scopedLog.Warningf(
```

Comment:
Usually we try to use the following format for log messages:

```
scopedLog.WithFields(logrus.Fields{
  "adaptor-limit": ...,
  "first-ifindex": ...,
}).Warning("single message that doesn't change")
```

This way, when integrating with log managers that understand the fields, they will correlate all messages of this type together regardless of what the fields or numbers are on any particular instance.

This would apply to all similar log statements in the PR.

Reply:
(Note that the extended log message would not need to be moved into the fields, I think that's just trying to help with sharing the same user recommendation across different log messages which probably makes sense as-is).

Reply:
indeed this looks much nicer, I updated it 👌